x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/index.tsx

```diff
@@ -77,6 +80,15 @@ export const EnterpriseSearchContentConfigured: React.FC<Required<InitialAppData
       <Route path={SEARCH_INDICES_PATH}>
         <SearchIndicesRouter />
       </Route>
+      <Route path={NEW_INDEX_SELECT_CONNECTOR_PATH} exact>
```

Comment:
This is the wrong place to do this: this is the base router, all the connector routes (and these are connector routes) should be under the connector router(`x-pack/plugins/serverless_search/public/application/components/connectors_router.tsx`). 

What we should do here is:
- Remove this route from the New Index router (you already did this 👍)
- Add this route to the Connectors Router (`x-pack/plugins/serverless_search/public/application/components/connectors_router.tsx`)
- Only have one instance of this route. All these three routes are identical, the router doesn't use query params to differentiate. We can just drop the `exact` from this route because SelectConnector doesn't have sub-routers.

Reply:
Good call! Just did it and did other route adjustments :)